crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -851,9 +852,22 @@ impl RoomInfo {
     }
 
     /// Handle the given redaction.
-    ///
-    /// Returns true if the event modified the info, false otherwise.
-    pub fn handle_redaction(&mut self, event: &OriginalSyncRoomRedactionEvent) {
+    pub fn handle_redaction(
+        &mut self,
+        event: &OriginalSyncRoomRedactionEvent,
+        _raw: &Raw<OriginalSyncRoomRedactionEvent>,
+    ) {
+        #[cfg(feature = "experimental-sliding-sync")]
+        if let Some(latest_event) = &mut self.latest_event {
+            if latest_event.event_id().as_deref() == Some(&*event.redacts) {
+                let room_version = self.base_info.room_version().unwrap_or(&RoomVersionId::V1);
+                match apply_redaction(&latest_event.event, _raw, room_version) {
+                    Some(redacted) => latest_event.event = redacted,
+                    None => self.latest_event = None,
```

Comment:
When that happens, do we want to look for an older `latest_event`?

Reply:
I don't think so.. Conceptually it's still redacted. If it weren't stored as a raw event, we'd probably have a `Redacted` variant here that we set the latest event to without attempting to actually apply the redaction properly. Maybe this is a good reason not to use a raw event actually, but I think we should discuss that separately from this PR.

Reply:
Sounds good!